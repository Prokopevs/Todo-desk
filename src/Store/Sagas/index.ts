import { fork } from 'redux-saga/effects';
import { authSaga } from './authSaga';
import { statusSaga } from './statusSaga';
import { taskSaga } from './taskSaga';

export default function* rootSaga() {
    yield fork(authSaga);
    yield fork(statusSaga);
    yield fork(taskSaga);
}