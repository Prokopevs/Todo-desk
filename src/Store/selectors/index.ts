export const selectError = state => state.errorMessageSlice
export const selectDnd = state => state.dndSlice
export const selectEditMode = state => state.editModeSlice
export const selectAuthorization = state => state.authorizationSlice
