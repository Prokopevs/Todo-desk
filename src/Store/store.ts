import editModeSlice from './reducers/editMode/slice';
import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'
import authorizationSlice from './reducers/authorization/slice'
import dndSlice from './reducers/dnd/slice'
import prioritySlice from './reducers/priority/slice';
import errorMessageSlice from './reducers/errorMessage/slice';
import rootSaga from './Sagas';

const saga = createSagaMiddleware()

export const store = configureStore({
    reducer: {
        authorizationSlice,
        dndSlice,
        prioritySlice,
        editModeSlice,
        errorMessageSlice,
    },
    middleware: [saga]
})
saga.run(rootSaga)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch