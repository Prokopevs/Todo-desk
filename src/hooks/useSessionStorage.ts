import { useState, useEffect } from "react";
 
function getStorageValue(key, defaultValue) {
    // getting stored value
    if (typeof window !== "undefined") {
        const saved = sessionStorage.getItem(key);
        const initial = saved !== null ? JSON.parse(saved) : defaultValue;
        return initial;
    }
}
 
export const useSessionStorage = (key, defaultValue) => {
    const [value, setValue] = useState(() => {
    return getStorageValue(key, defaultValue);
    });
 
    useEffect(() => {
        // storing input name
        sessionStorage.setItem(key, JSON.stringify(value));
    }, [key, value]);
 
    return [value, setValue];
};