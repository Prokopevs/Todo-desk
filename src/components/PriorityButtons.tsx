import React from "react"
import priorityArray from "../data/Desk/priorityArray"
import { useAppDispatch } from "../hooks/redux"
import { IPriorityButtons } from "../models/EditMode/IPriorityButtons"
import { setPriority } from "../Store/reducers/priority/slice"

const PriorityButtons: React.FC<IPriorityButtons> = ({ setChangePriority }) => {
    const dispatch = useAppDispatch()

    const changePriority = (index: number) => {
        dispatch(setPriority(index))
        setChangePriority(false)
    }

    return (
        <>
            <div role="priority"></div>
            {priorityArray.map((item, index) => (
                <li className="modalWindow__priority" key={index}>
                    <button
                        className={`block__button ${item.color} big mr1`}
                        onClick={() => changePriority(item.index)}
                    >
                        {item.description}
                    </button>
                </li>
            ))}
        </>
    )
}

export default PriorityButtons
