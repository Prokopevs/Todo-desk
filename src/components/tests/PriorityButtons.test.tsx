import { render, screen, fireEvent } from "@testing-library/react"
import React from "react"
import { Provider } from "react-redux"
import { store } from "../../Store/store"
import { MemoryRouter } from "react-router-dom"
import * as reactHooks from "../../hooks/redux"
import PriorityButtons from "../PriorityButtons"

jest.mock("../../hooks/redux")

const mockedSelector = jest.spyOn(reactHooks, "useAppSelector")
const mockedDispatch = jest.spyOn(reactHooks, "useAppDispatch")

beforeEach(() => {
    mockedDispatch.mockClear()
    mockedSelector.mockClear()
})

const setChangePriority = (...args: boolean[]) => {}

describe("PriorityButtons", () => {
    it("should create PriorityButtons component", () => {
        mockedSelector.mockReturnValue("")
        mockedDispatch.mockReturnValue(jest.fn())

        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <PriorityButtons setChangePriority={setChangePriority} />
                </MemoryRouter>
            </Provider>
        )

        expect(component).toMatchSnapshot()
    })

    it("should dispatch action", () => {
        mockedDispatch.mockReturnValue(jest.fn())
        mockedSelector.mockReturnValue("")

        render(
            <Provider store={store}>
                <MemoryRouter>
                    <PriorityButtons setChangePriority={setChangePriority} />
                </MemoryRouter>
            </Provider>
        )
        
        const priority = screen.getByRole("priority")
        fireEvent.click(priority)
        expect(mockedDispatch).toHaveBeenCalledTimes(1)
    })
})