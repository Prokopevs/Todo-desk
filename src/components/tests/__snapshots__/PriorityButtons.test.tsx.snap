// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriorityButtons should create PriorityButtons component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        role="priority"
      />
      <li
        class="modalWindow__priority"
      >
        <button
          class="block__button purple big mr1"
        >
          Very High
        </button>
      </li>
      <li
        class="modalWindow__priority"
      >
        <button
          class="block__button red big mr1"
        >
          High
        </button>
      </li>
      <li
        class="modalWindow__priority"
      >
        <button
          class="block__button green big mr1"
        >
          Normal
        </button>
      </li>
      <li
        class="modalWindow__priority"
      >
        <button
          class="block__button blue big mr1"
        >
          Low
        </button>
      </li>
      <li
        class="modalWindow__priority"
      >
        <button
          class="block__button lightblue big mr1"
        >
          Very Low
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <div
      role="priority"
    />
    <li
      class="modalWindow__priority"
    >
      <button
        class="block__button purple big mr1"
      >
        Very High
      </button>
    </li>
    <li
      class="modalWindow__priority"
    >
      <button
        class="block__button red big mr1"
      >
        High
      </button>
    </li>
    <li
      class="modalWindow__priority"
    >
      <button
        class="block__button green big mr1"
      >
        Normal
      </button>
    </li>
    <li
      class="modalWindow__priority"
    >
      <button
        class="block__button blue big mr1"
      >
        Low
      </button>
    </li>
    <li
      class="modalWindow__priority"
    >
      <button
        class="block__button lightblue big mr1"
      >
        Very Low
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
